### PR TITLE
feat: add ArgoCD

### LINKED ISSUE #7: Addon Request: ArgoCD
Request to add [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) as addon to MicroK8s.

### What Is Argo CD?
Argo CD is a declarative, GitOps continuous delivery tool for Kubernetes.

### Helm Chart

https://artifacthub.io/packages/helm/argo/argo-cd

### PR DESCRIPTION
**Pull request by dirien**
#### Please reference the issue this PR is fixing, or provide a description of the problem addressed.
Fixes #7 

Unfortunatly I could not test it on my local microk8s Installation (installed on multipass via snap)

I would be happy to test this, if someone point me on how to do this. The instructions in the `HACKING.md` did not work for me.

### CODE REVIEW

**Comment by neoaggelos** [comment]
Hi @dirien ! Thank you for this PR.

For some context, the addons repository is still under development and is expected to be officially released with MicroK8s 1.24. The easiest way to test already is to get a development build that uses this addons repository:

```
sudo snap install microk8s --classic --channel latest/edge/addons
# or,
sudo snap refresh microk8s --channel latest/edge/addons
```

You will then have the `/var/snap/microk8s/common/addons` which you can edit as HACKING.md suggests.

Thank you for contributing!

**Comment by dirien** [comment]
Wow thanks @neoaggelos for the feedback! This will help a lot to test the PR!

I set the PR meanwhile to Draft! Thanks again!

**Comment by dirien** [comment]
Now it could test it:

image

image

image

**Comment by dirien** [comment]
PS: Submitted the CLA form for contributing!

File: addons.yaml [hunk 1]
```diff
@@ -238,3 +238,10 @@ microk8s-addons:
       check_status: "daemonset.apps/fpga-operator"
       supported_architectures:
         - amd64
+
+    - name: "argocd"
+      description: "Argo CD is a declarative continuous deployment for Kubernetes."
+      version: "2.2.5"
+      check_status: "pod/argo-cd-server"
```
Comment (neoaggelos) [inline addons.yaml:245]:
```suggestion
      check_status: "deployment.apps/argo-cd-argcd-server"
```

I think the pod name is not correct here. This is the kubectl output for me:

```
# microk8s kubectl get pod,deploy -n argocd
NAME                                                         READY   STATUS    RESTARTS      AGE
pod/argo-cd-argocd-redis-58d4549574-fw68n                    1/1     Running   0             93s
pod/argo-cd-argocd-dex-server-5cb47f4755-hz2g5               1/1     Running   1 (83s ago)   93s
pod/argo-cd-argocd-server-5d8f4c89b7-ww9qh                   1/1     Running   0             93s
pod/argo-cd-argocd-application-controller-7b65595d88-kmmmf   1/1     Running   0             93s
pod/argo-cd-argocd-repo-server-7bd4cb988c-v2825              1/1     Running   0             93s

NAME                                                    READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/argo-cd-argocd-redis                    1/1     1            1           93s
deployment.apps/argo-cd-argocd-dex-server               1/1     1            1           93s
deployment.apps/argo-cd-argocd-server                   1/1     1            1           93s
deployment.apps/argo-cd-argocd-application-controller   1/1     1            1           93s
deployment.apps/argo-cd-argocd-repo-server              1/1     1            1           93s
```

With the suggested change, `microk8s status` shows ArgoCD as enabled:

```
# microk8s status --addon argocd
enabled
```

File: addons/argocd/disable [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+NAMESPACE_ARGOCD="argocd"
+
+KUBECTL="$SNAP/kubectl --kubeconfig=${SNAP_DATA}/credentials/client.config"
+HELM="$SNAP_DATA/bin/helm3 --kubeconfig=$SNAP_DATA/credentials/client.config"
+KUBECTL_DELETE_ARGS="--wait=true --timeout=180s --ignore-not-found=true"
+
+echo "Disabling ArgoCD"
+
+$HELM helm delete argo-cd -n $NAMESPACE_ARGOCD
```
Comment (neoaggelos) [inline addons/argocd/disable:15]: This is a typo I think
```suggestion
$HELM delete argo-cd -n $NAMESPACE_ARGOCD
```

File: addons/argocd/disable [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+$KUBECTL delete customresourcedefinition applications.argoproj.io \
+                argocdextensions.argoproj.io  \
+                appprojects.argoproj.io
+
+$KUBECTL delete $KUBECTL_DELETE_ARGS namespace "$NAMESPACE_ARGOCD" > /dev/null 2>&1 || true
+
+echo "ArgoCD disabled"
```
Comment (neoaggelos) [inline addons/argocd/disable:23]: nit; add an empty new-line

**Review by neoaggelos — CHANGES_REQUESTED** [review]
Thanks! Looking good, but with some small issues! Please see the suggested fixes!

File: addons.yaml (hunk 1 above)
Comment (dirien) [reply addons.yaml:245]: Wow thanks for that one! good finding!

File: addons/argocd/disable (hunk 2 above)
Comment (dirien) [reply addons/argocd/disable:15]: classic type!

File: addons/argocd/disable (hunk 3 above)
Comment (dirien) [reply addons/argocd/disable]: done! 👍

**Comment by dirien** [comment]
image

image


Should work fine now! Thanks @neoaggelos for the feedback!

File: tests/validators.py [hunk 4]
```diff
@@ -535,6 +535,12 @@ def validate_openfaas():
     wait_for_pod_state("", "openfaas", "running", label="app=gateway")
 
 
+def validate_argocd():
+    """
+    Validate argocd
+    """
+    wait_for_pod_state("", "argocd", "running", label="app.kubernetes.io/component=server")
+
```
Comment (neoaggelos) [inline tests/validators.py:543]: Final thing, add an extra new line to make `black` happy

**Review by neoaggelos — COMMENTED** [review]
Almost there, please one final fix to make CI happy

File: tests/validators.py (hunk 4 above)
Comment (dirien) [reply tests/validators.py:543]: done 😄

**Comment by dirien** [comment]
Hi @neoaggelos i am note sure what is wrong. I am not a python dev, but if you tell me how to run the validation in my local dev, I will test it there.

**Pull request merged**